runtimes/cloudformation/cfnpatcher/util.go

```diff
@@ -7,16 +7,27 @@ import (
 	"github.com/Jeffail/gabs/v2"
 )
 
-func getTags(template *gabs.Container) map[string]string {
-	tags := make(map[string]string)
-	if template.Exists("Properties", "Tags") {
-		for _, tag := range template.S("Properties", "Tags").Children() {
-			if tag.Exists("Key") && tag.Exists("Value") {
-				tags[tag.S("Key").Data().(string)] = tag.S("Value").Data().(string)
+func getOptTags(template *gabs.Container) map[string]string {
+	optTags := make(map[string]string)
+	if !template.Exists("Properties", "Tags") {
+		return optTags
+	}
+	for _, tag := range template.S("Properties", "Tags").Children() {
+		if tag.Exists("Key") && tag.Exists("Value") {
+			k, ok := tag.S("Key").Data().(string)
+			if !ok {
+				panic(fmt.Errorf("tag has an unsupported key type: %s", tag.String()))
+			}
```

Comment:
Hmm, if this can happen in the real world, maybe we should silently (or with a warning) skip this tag instead of panicking?

Reply:
not sure...in that case we would not honore the opt config and the resulting patch would not be correct

Reply:
Okay, let's leave it as is and see if anybody complains ;) (I assume there's no clean and easy way to resolve the `!Ref`s and avoid the problem in the first place)